config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,20 +924,24 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
+        // this has two possible outcomes depending on whether
+        // we resolve and memoize a first or b first. currently
+        // java 8's hash table makes it resolve OK, but
```

Comment:
FWIW the 'config file round-tripping' stuff that we are playing around with based on the commentary in issue #149 might need LinkedHashMap for ordering as well, and we'd be really really happy with a solution that worked in Java 7 as well as Java 8  :)  I realize that I probably don't get a vote here, but am interested nonetheless :)